src/setupProxy.js

```diff
@@ -0,0 +1,12 @@
+const { createProxyMiddleware } = require('http-proxy-middleware')
+
+
+module.exports = function(app) {
+  app.use(
+    '/api',
+    createProxyMiddleware({
+      target: 'http://localhost:8080', // CBAS Service
+      changeOrigin: true
+    })
+  )
+}
```

Comment:
Was this for the in-UI routing thing? I would guess we don't need this any more?

Reply:
We still need this to resolve the CORS error that we get when running local UI with local CBAS Service

Reply:
I remember running into this and fixing it with an annotation in CBAS itself (which I think is where the error was coming from) along the lines of "allow requests from localhost". I'm not sure which is neater, but in either case I think it's probably important not to leave it activated accidentally in production

Reply:
Yes I had tried adding `@CrossOrigin` annotation to the /POST endpoint method handler in CBAS, but the guidance by App Sec (see [this Slack thread](https://broadinstitute.slack.com/archives/C2B7JFJ4S/p1657053506352529?thread_ts=1657036114.427909&cid=C2B7JFJ4S)) was not to have this annotation checked in as it can introduce security vulnerabilities. Instead they had suggested adding the local proxy which is only activated in development mode. I think the article also mentions that this is activated in development only.